Add libharu

xylar opened the pull request:
@hobu asked me to submit this as a replacement for #10375, which he will close.

This library is needed by `vtk`, perhaps among other packages.


<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with
a PR or once it's ready for review, please let the right people know.
There are language-specific teams for reviewing recipes.

Currently available teams are:
- python `@conda-forge/help-python`
- python/c hybrid `@conda-forge/help-python-c`
- r `@conda-forge/help-r`
- java `@conda-forge/help-java`
- nodejs `@conda-forge/help-nodejs`
- c/c++ `@conda-forge/help-c-cpp`
- perl `@conda-forge/help-perl`
- Julia `@conda-forge/help-julia`
- ruby `@conda-forge/help-ruby`

If your PR doesn't fall into those categories please contact
the full review team `@conda-forge/staged-recipes`.

Due to GitHub limitations first time contributors to conda-forge are unable
to ping these teams. You can [ping the team](https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team) using a special command in
a comment on the PR to get the attention of the `staged-recipes` team. You can
also consider asking on our [Gitter channel](https://gitter.im/conda-forge/conda-forge.github.io)
or on our [Keybase chat](https://keybase.io/team/condaforge.chat)
if your recipe isn't reviewed promptly.
-->

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/libharu```) and found it was in an excellent condition.

xylar commented on recipes/libharu/meta.yaml:
```diff
@@ -0,0 +1,45 @@
+
+extra:
+  recipe-maintainers:
+    - hobu
```
@hobu, are you still willing to be a maintainer?

xylar commented:
Hoping to address request https://github.com/conda-forge/staged-recipes/issues/5855 and partially address issue https://github.com/conda-forge/vtk-feedstock/issues/55

xylar commented:
@conda-forge/help-c-cpp, I'm not sure why my patch doesn't work in appveyor but works fine in azure under win.  

If that's a problem, I could use some help figuring out what might be wrong with the patch.  

If it's an appveryor issue, maybe we can move on to a review?

xhochy commented on recipes/libharu/bld.bat:
```diff
@@ -0,0 +1,16 @@
+mkdir build
+cd build
+
+cmake -G "NMake Makefiles" ^
```
Can you use `ninja` or `jom` to make use of parallelisation?

xylar replied:
Thanks very much for the suggestion, I have switched to `ninja`, as you recommend.

xhochy commented on recipes/libharu/meta.yaml:
```diff
@@ -0,0 +1,53 @@
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - cmake
```
Please add `- make  # [unix]` here. Otherwise you could also just settle for `- ninja` on all platforms for maximum resource utilisation.

xylar replied:
I have, indeed, switched to `ninja`, so we'll see how the Windows and OSX builds do, but this worked well for Linux.

xhochy commented on recipes/libharu/meta.yaml:
```diff
@@ -0,0 +1,53 @@
+  host:
+    - zlib
+    - libpng
+    - boost
```
Do you need `Boost.Python`? Otherwise you should depend on `boost-cpp`.

xylar replied:
I'm not sure why @hobu added `boost` at all.  I have removed it and things build fine.

xhochy commented on recipes/libharu/meta.yaml:
```diff
@@ -0,0 +1,53 @@
+  license: Zlib
+  license_family: OTHER
+  license_file: LICENSE.txt
+  summary: Simple, fast, extensible JSON encoder/decoder for Python
```
You need to update the description.

xylar replied:
Oops!  What a silly oversight on my part.  I have moved up the description, which is short enough to be a summary.

xylar commented:
@xhochy, thank you very much for the review.  I believe I have addressed your concerns, but still waiting on CI.  I'll let you know when it is ready for you to follow up.

xylar commented:
@xhochy, okay I think it's ready for a second review.  Thanks again.

xylar commented:
@xhochy, thank you so much!
